File: golang/1356.sort-integers-by-the-number-of-1-bits.go
import (
	"math/bits"
)

func sortByBits(arr []int) []int {
	sort.Slice(arr, func(i, j int) bool {
		// |--- ones count --| |----- bits ------|
		// 0000 0000 0000 0110 0000 0001 1010 1011
		a := bits.OnesCount(uint(arr[i]))<<16 | arr[i]
		b := bits.OnesCount(uint(arr[j]))<<16 | arr[j]
		return a < b
	})

	return arr
}

File: cpp/1356.sort-integers-by-the-number-of-1-bits.cpp
#include <bit>

class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        sort(arr.begin(), arr.end(), [](int a, int b) -> bool {
            // <--- bits count --> <------ bits ----->
            // 0000 0000 0000 0110 0000 0000 1101 1011
            int v1 = std::popcount(uint(a)) << 16 | a;
            int v2 = std::popcount(uint(b)) << 16 | b;

            return v1 < v2;
        });

        return arr;
    }
};